### ISSUE TITLE
Authenticated proxies with '#' symbol not handled properly

### ISSUE THREAD

**Issue by kenoung** [scrapinghub/splash#680]
https://github.com/scrapinghub/splash/blob/23d4bcf0bc0fac031b190c8827568cd842fb3599/splash/proxy.py#L179

Using urlparse will introduce problems for authenticated proxies e.g. http://username:p@ssw#rd@vpn.com:90

```
>>> from urllib.parse import urlparse
>>> urlparse('http://username:p@ssw#rd@vpn.com:90')
ParseResult(scheme='http', netloc='username:passw', path='', params='', query='', fragment='d@vpn.com:90')
```

A possible fix might be to use `_parse_proxy` from `urllib.request` as done in the [scrapy httpproxy middleware](https://github.com/scrapy/scrapy/blob/master/scrapy/downloadermiddlewares/httpproxy.py).
```
>>> from urllib.request import _parse_proxy
>>> _parse_proxy('http://username:p@ssw#rd@vpn.com:90')
('http', 'username', 'p@ssw#rd', 'vpn.com:90')
```

**Comment by kmike** [comment]
Yeah, a good point. Proxy profiles or request:set_proxy don't seem to have the same issue (so you can use them as a workaround), but passing "proxy" in an argument is affected.
